### ISSUE TITLE
Incorrect filename of downloaded file

### ISSUE THREAD

**Issue by akoval** [codeborne/selenide#734]
## Incorrect filename after downloading
Filename of the downloaded file is the same as href. So 
`href="myfileblabla.txt?sessioncookie=12345%2323`
will produce file with name "myfileblabla.txt?sessioncookie=12345%2323"

## Details
Need to extract filename from href until `?`

## Tell us about your environment
* **Selenide Version:** 4.11.4
* **Chrome\Firefox\IE Version:** 66
* **OS Version:** Windows

**Comment by akoval** [comment]
this issue can raise issues on Windows. Not all filenames are valid on Windows OS

**Comment by rosolko** [comment]
@akoval Can you please specify what type on `fileDownload` do you use in `Configuration`?

**Comment by akoval** [comment]
@rosolko I use only that

`
        Configuration.browser = "chrome";
        Configuration.timeout = 10000;
`

Should I setup smth specific?

**Comment by rosolko** [comment]
@akoval Got it. No, you don't. Thanks for information.

**Comment by akoval** [comment]
@rosolko problem is still actual.

I found the place where it should construct filename.

`protected String getFileName(String fileToDownloadLocation, HttpResponse response) throws MalformedURLException {
    for (Header header : response.getAllHeaders()) {
      Optional<String> fileName = httpHelper.getFileNameFromContentDisposition(header.getName(), header.getValue());
      if (fileName.isPresent()) {
        return fileName.get();
      }
    }

    log.info("Cannot extract file name from http headers. Found headers: ");
    for (Header header : response.getAllHeaders()) {
      log.info(header.getName() + '=' + header.getValue());
    }

    return new URL(fileToDownloadLocation).getFile().replaceFirst("[/\\\\]", "");
  }`

I see following message in the logs:
`"Cannot extract file name from http headers. Found headers: "`

So looks like your fix is working only for content-disposition case

**Comment by akoval** [comment]
[demo.zip](https://github.com/codeborne/selenide/files/2223797/demo.zip)

I uploaded demo project that contains 1 html page with link to the file with some query params.

Open http://localhost:8080/1.html. Link is `http://localhost:8080/test.txt?www=1111&rrr=2222`

Test:

```
Configuration.browser = "chrome";
Configuration.timeout = 10000;
open("http://localhost:8080/1.html");
$("#link").download();
```

When I click the link I have no any Content-Disposition header

**Comment by asolntsev** [comment]
@akoval @rosolko Thank you. Will be released in Selenide 4.12.4
